File: typescript/src/errors/ConfigurationError.ts
import {
  ErrorCode,
  ParmanaError,
} from "./ParmanaError.js";

/**
 * Raised when SDK configuration is invalid.
 */
export class ConfigurationError extends ParmanaError {
  constructor(
    message: string,
    options?: {
      requestId?: string;
      cause?: unknown;
    },
  ) {
    super({
      code: ErrorCode.CONFIGURATION_ERROR,

      message,

      ...(options?.requestId !== undefined && {
        requestId: options.requestId,
      }),

      ...(options?.cause !== undefined && {
        cause: options.cause,
      }),
    });

    this.name = "ConfigurationError";
  }
}

File: python/parmana/errors/configuration_error.py
from __future__ import annotations

from .api_error import ParmanaError


class ConfigurationError(ParmanaError):
    """
    Raised when SDK configuration is invalid (e.g. a missing Runtime
    endpoint). Mirrors typescript/src/errors/ConfigurationError.ts.
    """

    def __init__(
        self,
        message: str,
        *,
        request_id: str | None = None,
    ) -> None:
        super().__init__(
            message,
            code="CONFIGURATION_ERROR",
            request_id=request_id,
        )
